lua/neogit/popups/fetch.lua

```diff
@@ -0,0 +1,57 @@
+local popup = require("neogit.lib.popup")
+local git = require("neogit.lib.git")
+local a = require("plenary.async")
+local notif = require("neogit.lib.notification")
+local fetch_lib = require("neogit.lib.git.fetch")
+local input = require("neogit.lib.input")
+local status = require("neogit.status")
+
+local M = {}
+
+local function fetch_from(name, remote, branch, args)
+  notif.create("Fetching from " .. name)
+  local res = fetch_lib.fetch_interactive(remote, branch, args)
+
+  if res and res.code == 0 then
+    a.util.scheduler()
+    notif.create("Fetched from " .. name)
+    vim.cmd("do <nomodeline> User NeogitFetchComplete")
+  end
+end
+
+function M.create()
+  local p = popup
+    .builder()
+    :name("NeogitFetchPopup")
+    :switch("p", "prune", "Prune deleted branches", false)
+    :switch("t", "tags", "Fetch all tags", false)
+    :action("p", "Fetch from pushremote", function(popup)
+      fetch_from("pushremote", "origin", status.repo.head.branch, popup:get_arguments())
+    end)
+    :action("u", "Fetch from upstream", function(popup)
+      local upstream = git.branch.get_upstream()
+      if not upstream then
+        return
+      end
+
+      fetch_from(upstream.remote, upstream.remote, "", popup:get_arguments())
+    end)
+    :action("a", "Fetch from all remotes", function(popup)
+      local args = popup:get_arguments()
+      table.insert(args, "--all")
+
+      fetch_from("all remotes", "", "", args)
+    end)
+    :action("e", "Fetch from elsewhere", function(popup)
+      local remote = input.get_user_input("remote: ")
+      local branch = git.branch.prompt_for_branch()
```

Comment:
FYI there is a non-blocking and much prettier split based branch selection now

https://github.com/TimUntersberger/neogit/blob/efc1764060e64cc8e81687319d5223b3246e5871/lua/neogit/popups/merge.lua#L26-L37

Reply:
IMO its easier to leave as is - I completely replaced the branch picker interface with telescope. For simplicity's sake I made it a required dependency, so we don't need to maintain two parallel implementations. Updating this PR just adds a conflict to resolve 🤷

Reply:
Good point, was just a consideration, and is not strictly necessary to merge. There are many other places where the old branch picker is used.

Reply:
Totally fair - the new branch picker _is_ an improvement. I was thinking about it on my way home from work and it could be a good strategy to roll out the dependency - do something like:
```lua
local ok, tele = pcall(require, "telescope")
if ok then
  -- Use telescope picker
else
  -- Print a notice to add dependency and fallback to OG picker
end
```